controller/backing_image_controller.go

```diff
@@ -310,22 +345,604 @@ func (bic *BackingImageController) syncBackingImage(key string) (err error) {
+		if exists {
+			if biStatus.State == longhorn.BackingImageStateReady {
+				readyV2CopyDiskUUIDs = append(readyV2CopyDiskUUIDs, v2DiskUUID)
+			}
+			if backingImageInProgress(biStatus.State) {
+				log.Infof("There is one v2 backing image copy in progress, prepare the backing image one at a time", bi.Name)
+				return nil
+			}
+		} else {
+			requireV2CopyDiskUUIDs = append(requireV2CopyDiskUUIDs, v2DiskUUID)
+		}
+	}
+
+	if len(readyV2CopyDiskUUIDs) == 0 {
+		log.Infof("Only sync the backing image to other disks when there is one ready v2 backing image copy")
+		return nil
+	}
+
+	sourceV2DiskUUID := readyV2CopyDiskUUIDs[0]
+	// Only sync one backing image copy at a time, so we can control the concurrent limit of syncing for the backing image manager.
+	for _, v2DiskUUID := range requireV2CopyDiskUUIDs {
+		if bic.v2CopyBackoff.IsInBackOffSinceUpdate(v2DiskUUID, time.Now()) {
+			log.Debugf("Skip syncing backing image copy to disk %v immediately since it is still in the backoff window", v2DiskUUID)
+			continue
+		}
+
+		node, _, err := bic.ds.GetReadyDiskNodeRO(v2DiskUUID)
+		if err != nil {
+			bic.v2CopyBackoff.Next(v2DiskUUID, time.Now())
+			return errors.Wrapf(err, "failed to get the ready disk node for disk %v", v2DiskUUID)
+		}
+
+		instanceManager, err := bic.ds.GetRunningInstanceManagerByNodeRO(node.Name, longhorn.DataEngineTypeV2)
+		if err != nil {
+			bic.v2CopyBackoff.Next(v2DiskUUID, time.Now())
+			return errors.Wrapf(err, "failed to get the default instance manager for node %v", node.Name)
+		}
+		engineClientProxy, err := engineapi.NewEngineClientProxy(instanceManager, log, bic.proxyConnCounter)
+		if err != nil {
+			bic.v2CopyBackoff.Next(v2DiskUUID, time.Now())
+			return errors.Wrapf(err, "failed to get the engine client proxy for instance manager %v", instanceManager.Name)
+		}
+
+		log.Infof("Syncing the backing image to disk %v for backing image %v from disk: %v", v2DiskUUID, bi.Name, sourceV2DiskUUID)
+		srcNode, _, err := bic.ds.GetReadyDiskNodeRO(sourceV2DiskUUID)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get the ready disk node for disk %v", v2DiskUUID)
+		}
+		srcInstanceManager, err := bic.ds.GetRunningInstanceManagerByNodeRO(srcNode.Name, longhorn.DataEngineTypeV2)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get the default instance manager for node %v", node.Name)
+		}
+
+		// Create the backing image by syncing the backing image data from the SPDK server inside the instance manager holding the source disk
+		_, err = engineClientProxy.SPDKBackingImageCreate(bi.Name, bi.Status.UUID, v2DiskUUID, bi.Status.Checksum, fmt.Sprintf("%v:%v", srcInstanceManager.Status.IP, engineapi.InstanceManagerSpdkServiceDefaultPort), sourceV2DiskUUID, uint64(bi.Status.Size))
+		engineClientProxy.Close()
+		if err != nil {
+			bic.v2CopyBackoff.Next(v2DiskUUID, time.Now())
+			return errors.Wrapf(err, "failed to create the v2 backing image on diskUUID %v", v2DiskUUID)
```

Comment:
Suggest abstracting the codes into a function and close the client in a defer function.

Reply:
fixed.